plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerDelegate.java

```diff
@@ -131,23 +134,33 @@ IStatus checkConflictingServiceIds(IModule[] current, IModule[] toBeAdded,
     // verify that we do not have conflicting modules by service id
     Map<String, IModule> currentServiceIds = new HashMap<>();
     for (IModule module : current) {
-      String moduleServiceId = getServiceId(module);
-      if (currentServiceIds.containsKey(moduleServiceId)) {
-        // should never happen: we have a conflict within the already-defined modules
-        return StatusUtil.error(this, Messages.getString("SERVICE_CONFLICTS", //$NON-NLS-1$
-            module.getName(), currentServiceIds.get(moduleServiceId).getName(), moduleServiceId));
+      // ignore deleted modules / closed projects
+      if (module.exists()) {
+        String moduleServiceId = getServiceId(module);
+        if (currentServiceIds.containsKey(moduleServiceId)) {
+          // should never happen: we have a conflict within the already-defined modules
+          return StatusUtil.error(this, Messages.getString("SERVICE_CONFLICTS", //$NON-NLS-1$
+              module.getName(), currentServiceIds.get(moduleServiceId).getName(), moduleServiceId));
+        }
+        currentServiceIds.put(moduleServiceId, module);
       }
-      currentServiceIds.put(moduleServiceId, module);
     }
     if (toBeRemoved != null) {
       for (IModule module : toBeRemoved) {
-        String moduleServiceId = getServiceId(module);
-        // could verify that: serviceIds.containsKey(moduleServiceId)
-        currentServiceIds.remove(moduleServiceId);
+        // always ok to remove deleted modules / closed projects
+        if (module.exists()) {
+          String moduleServiceId = getServiceId(module);
+          // could verify that: serviceIds.containsKey(moduleServiceId)
```

Comment:
Why this comment? Should we do this?

Reply:
Hmm, dunno: that comment was from long ago.  I suspect it meant that we could confirm that the module was in the list of current modules.